Sources/Subs.php

```diff
@@ -7040,4 +7040,78 @@ function sentence_list($list)
 	return strtr($format, $replacements);
 }
 
+/**
+ * Truncate an array to a specified length
+ *
+ * @param array $array The array to truncate
+ * @param int $max_length The upperbound on the length
+ * @param int $deep How levels in an multidimensional array should the function take into account.
+ * @return array The truncated array
+ */
+function truncate_array($array, $max_length = 1900, $deep = 3)
+{
+    $array = (array) $array;
+
+    $curr_length = 0;
+    $curr_length = array_length($array);
```

Comment:
This should probably be `array_length($array, $deep)` in order to ensure that both functions are working to the same depth.

Reply:
agree